src/store/modules/segmentationModule.js

```diff
@@ -0,0 +1,1411 @@
+ * @property {Uint16Array} pixelData A 2D view on a section of the parent `Labelmap3D`'s `buffer`.
+ * @property {number[]} segmentsOnLabelmap An array of segments present in the `pixelData`.
+ * @property {boolean} invalidated Whether the data has been changed recently, and a new `ImageBitmap` should be created
+ *                                 for the segmentation fill should this `labelmap2D` be rendered.
+ */
+
+/**
+ * GetMetadata - Returns the metadata object for a partiular segment if
+ * segmentIndex is specified, otherwise returns an array of all segment metadata
+ * for the labelmap.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
+ * @param  {number} [labelmapIndex]    If undefined, defaults to the active
+ *                                     labelmap index.
+ * @param  {number} [segmentIndex]     The segment index.
+ * @returns {Object|Object[]}          A metadata object or an array of
+ *                                     metadata objects.
+ */
+function getMetadata(elementOrEnabledElementUID, labelmapIndex, segmentIndex) {
+  const element = _getEnabledElement(elementOrEnabledElementUID);
+
+  if (!element) {
+    return;
+  }
+
+  const stackState = getToolState(element, 'stack');
+  const stackData = stackState.data[0];
+  const firstImageId = stackData.imageIds[0];
+
+  const brushStackState = state.series[firstImageId];
+
+  if (!brushStackState) {
+    logger.warn(`brushStackState is undefined`);
+
+    return;
+  }
+
+  if (labelmapIndex === undefined) {
+    labelmapIndex = brushStackState.activeLabelmapIndex;
+  }
+
+  if (!brushStackState.labelmaps3D[labelmapIndex]) {
+    logger.warn(`No labelmap3D of labelmap index ${labelmapIndex} on stack.`);
+
+    return;
+  }
+
+  const labelmap3D = brushStackState.labelmaps3D[labelmapIndex];
+
+  if (segmentIndex === undefined) {
+    return labelmap3D.metadata;
+  }
+
+  return labelmap3D.metadata[segmentIndex];
+}
+
+/**
+ * GetLabelmaps3D - Returns the `Labelmap3D` objects associated with the series displayed
+ * in the element, the `activeLabelmapIndex` and the `currentImageIdIndex`.
+ *
+ * @param  {HTMLElement|string} elementOrEnabledElementUID   The cornerstone enabled
+ *                                                    element or its UUID.
+ * @returns {Object}              An object containing `Labelmap3D` objects,
+ *                                the `activeLabelmapIndex` amd the `currentImageIdIndex`.
+ */
+function getLabelmaps3D(elementOrEnabledElementUID) {
+  const element = _getEnabledElement(elementOrEnabledElementUID);
+
+  if (!element) {
+    return;
+  }
+
+  const stackState = getToolState(element, 'stack');
+  const stackData = stackState.data[0];
+
+  const firstImageId = stackData.imageIds[0];
+  const brushStackState = state.series[firstImageId];
```

Comment:
It might be nice to make this reusable since we call it in a few places, and may retrieve the brushStackState differently in the future.

Reply:
I agree, I'll do a bit of cleanup today for this module.